### ISSUE TITLE
Problem with out() function when you combine it with the name of edge using underscore characters and dot syntax

### ISSUE THREAD

**Issue by healiseu** [orientechnologies/orientdb#6155]
#### Create a graph database with lightweight edges
``` JAVASCRIPT
ALTER DATABASE custom useLightweightEdges=true
```

#### Create Classes and Properties

``` JAVASCRIPT
create class H extends V
create class isHyperedgeOf extends E

create class WORD extends V
create class INT extends V
create class REAL extends V

create property WORD.dval STRING
create property INT.dval INTEGER
create property REAL.dval DOUBLE

create class NameGR extends V
create class NameEN extends V
create class Enum extends V
create class Price extends V

create property NameGR.dptr LINK WORD
create property NameEN.dptr LINK WORD
create property Enum.dptr LINK INT
create property Price.dptr LINK REAL
```

#### Create Nodes
``` JAVASCRIPT
create vertex H

create vertex WORD set dval="χψ"
create vertex WORD set dval="FG"
create vertex INT set dval=1
create vertex REAL set dval=1.9

In my setup 14:0, 14:1, 15:0, 16:0 have been created

create vertex NameGR set dptr=14:0
create vertex NameEN set dptr=14:1
create vertex Enum set dptr=15:0
create vertex Price set dptr=16:0
```

At this step 17:0, 18:0, 19:0, 20:0 have been created and they are linked to
14:0, 14:1, 15:0, 16:0 respectively 

#### Create Edges
``` JAVASCRIPT
create edge isHyperedgeOf from 12:0 to 17:0
create edge isHyperedgeOf from 12:0 to 18:0
create edge isHyperedgeOf from 12:0 to 19:0
create edge isHyperedgeOf from 12:0 to 20:0
```

#### Query
``` JAVASCRIPT
select out_isHyperedgeOf from 12:0
this returns the list of adjacent outgoing vertices and the result is as expected
17:0, 18:0, 19:0, 20:0

and the same result is obtained by
select out() from 12:0
```

#### Problem
``` JAVASCRIPT
select out().dptr from 12:0
#14:0  #14:1  #15:0  #16:0

select out().dptr.dval from 12:0
["χψ","FG",1,1.9]

This is as expected, but if I use

select out_isHyperedgeOf.dptr from 12:0
#16:0  #15:0  #14:0  #14:1

select out_isHyperedgeOf.dptr.dval from 12:0
[1.9,1,"χψ","FG"]

As you see the returned list has lost its order of elements
```

#### Is that a Bug ?
....when you combine out() function with the name of edge using underscore characters.....

#### Solution
``` JAVASCRIPT
select out('isHyperedgeOf').dptr.dval from 12:0
["χψ", "FG", 1, 1.9]
```

### Runninng Mode
- [X ] Embedded, using PLOCAL access mode

### OrientDB Version
- [X] v2.1.16

### Operating System
- [X ] Windows

**Comment by luigidellaquila** [comment]
Hi @healiseu 

I fixed this in both 2.1.x and develop branches, the fix will be available in 2.1.18 and 2.2.1

Thanks

Luigi

**Comment by healiseu** [comment]
:+1:
